src/lib/extend.ts

```diff
@@ -184,27 +186,65 @@ const modernExtend = {
     numeric: (args: {
         name: string, cluster: string | number, attribute: string | {id: number, type: number}, description: string,
         zigbeeCommandOptions?: {manufacturerCode: number}, readOnly?: boolean, unit?: string,
+        valueMin?: number, valueMax?: number, valueStep?: number, scale?: number,
     }): ModernExtend => {
         const {name, cluster, attribute, description, zigbeeCommandOptions} = args;
         const attributeKey = isString(attribute) ? attribute : attribute.id;
 
-        let expose = new Numeric(name, args.readOnly ? access.STATE_GET : access.ALL).withDescription(description);
+        let expose = new Numeric(name, args.readOnly ? access.STATE_GET : access.ALL).withDescription(description)
+            .withValueMin((args.valueMin) ? args.valueMin : 0)
+            .withValueMax((args.valueMax) ? args.valueMax : 100)
+            .withValueStep((args.valueStep) ? args.valueStep : 1);
         if (args.unit) expose = expose.withUnit(args.unit);
 
         const fromZigbee: Fz.Converter[] = [{
             cluster,
             type: ['attributeReport', 'readResponse'],
             convert: (model, msg, publish, options, meta) => {
                 if (attributeKey in msg.data) {
-                    return {[expose.property]: msg.data[attributeKey]};
+                    const val = (args.scale) ? toNumber(msg.data[attributeKey]) / toNumber(args.scale) : msg.data[attributeKey];
+                    return {[expose.property]: val};
                 }
             },
         }];
 
         const toZigbee: Tz.Converter[] = [{
             key: [name],
             convertSet: args.readOnly ? undefined : async (entity, key, value, meta) => {
-                const payload = isString(attribute) ? {[attribute]: value} : {[attribute.id]: {value, type: attribute.type}};
+                const val = (args.scale) ? toNumber(value) * toNumber(args.scale) : value;
+                const payload = isString(attribute) ? {[attribute]: val} : {[attribute.id]: {value: val, type: attribute.type}};
+                await entity.write(cluster, payload, zigbeeCommandOptions);
+                return {state: {[key]: value}};
+            },
+            convertGet: async (entity, key, meta) => {
+                // @ts-expect-error TODO fix zh type
+                await entity.read(cluster, [attributeKey], zigbeeCommandOptions);
+            },
+        }];
+        return {exposes: [expose], fromZigbee, toZigbee, isModernExtend: true};
+    },
+    binary: (args: {
+        name: string, valueOn: string | number | boolean, valueOff: string | number | boolean,
+        cluster: string | number, attribute: string | {id: number, type: number}, description: string,
+        zigbeeCommandOptions?: {manufacturerCode: number}, readOnly?: boolean,
+    }): ModernExtend => {
+        const {name, valueOn, valueOff, cluster, attribute, description, zigbeeCommandOptions} = args;
```

Comment:
we should have 2 valueOns here, one for the expose (instead of hardcoded `ON`/`OFF` and for the zigbee value). Something like `valueOn` and `zigbeeValueOn`?